docs/debugging.md

```diff
@@ -0,0 +1,39 @@
+## Debugging Tutorial
+
+### For debugging purposes the following must be installed :
+
+[Visual Studio Code](https://code.visualstudio.com/)
+[Node.js](https://www.npmjs.com/) - to setup node.js you may follow the instructions in [Environment Setup](https://github.com/Seneca-CDOT/telescope/blob/master/docs/environment-setup.md)
+
+### Setting up VS Code and Launching Telescope :
+
+To begin debugging one must select the "Run View" tab. This will display a menu that for debugging purposes.
+
+![VS Run View Screenshot](./images/VS-Run-View-Screenshot.png)
+Visual Studio Code will automatically try to detect the debug environment but in the case that it fails you must select "Node.js".
```

Comment:
I think this needs an extra newline:

![image](https://user-images.githubusercontent.com/23108901/100531117-5ee1ab00-31c8-11eb-9845-fcfd6b983d59.png)

Reply:
@manekenpix I've fixed that new line, I also added an additional screenshot to display all the features that can be utilized with VS code.

I now see that the code is failing due to a new feature that's been added... If you give me a hint ( as I don't see anything in the docs )  on how to resolve this I'll be more than happy to fix it ! 

Thanks !

Reply:
@mamadou-diallo I think rebasing on upstream master will fix the issue

Reply:
@mamadou-diallo also, I saw that you added VScode features, but I meant telescope's debugging options. If you take a look at the screenshot I posted in my first comment, you can see how you can choose a debugging option depending on what you want to do:
- Launch Telescope (for debugging the Telescope app)
- Launch Auto Deployment (for debugging the autodeployment server)
- Launch All Tests (for debugging our test set)
- Launch Opened Test File (for debugging an opened test in VScode)

For detailed info about these options, check out our [launch,json](https://github.com/Seneca-CDOT/telescope/blob/master/.vscode/launch.json) in the `.vscode` folder.

I think it'd be more interesting if the document this PR is adding focused on the 4 different debugging options that Telescope has (which is what's really helpful for developers) more than explaining VScode features.

Let me know what you think.

Reply:
@manekenpix Oops ! I didn't get what you meant initially but I don't mind doing that correction.